### PR TITLE
Re-enable Tests

### PR DESCRIPTION
**Pull request by T045T**
They're still far from comprehensive, but they compile (and the ones I could test succeed) now.

Some of them require KShoot beatmaps that I don't have, so I didn't test them.

### CODE REVIEW

File: Audio/src/Audio.cpp [hunk 1]
```diff
@@ -103,6 +102,11 @@ void Audio::Mix(float* data, uint32& numSamples)
 
 void Audio::start()
 {
+	std::scoped_lock l(lock);
+	if (m_sampleBuffer || limiter) {
+		Logf("Audio.start() called twice!", Logger::Error);
```
Comment (BonusPlay) [inline Audio/src/Audio.cpp]: Replace with `Log`, since `Logf` formats the output.

File: GUI/src/LayoutBox.cpp [hunk 2]
```diff
@@ -77,7 +77,6 @@ Vector2 LayoutBox::GetDesiredSize(GUIRenderData rd)
 
 LayoutBox::Slot* LayoutBox::Add(GUIElement element)
 {
-	bool found = false;
 	for (auto it = m_children.begin(); it != m_children.end(); it++)
```
Comment (BonusPlay) [inline GUI/src/LayoutBox.cpp]: while you're at it, could you replace this with `for (auto& it : m_children)` ?

File: Main/src/Camera.cpp [hunk 3]
```diff
@@ -166,7 +166,7 @@ RenderState Camera::CreateRenderState(bool clipped)
 
 	m_pitch = base_pitch - pitchOffset;
 
-	float d0 = VectorMath::Dot(Vector3(0.0f, 0.0f, 0.0f), cameraDir) + offset;
+	// float d0 = VectorMath::Dot(Vector3(0.0f, 0.0f, 0.0f), cameraDir) + offset;
```
Comment (BonusPlay) [inline Main/src/Camera.cpp]: This is suspiciously not used, any idea why it works without it?

File: Audio/src/Audio.cpp [hunk 4]
```diff
@@ -32,42 +30,43 @@ void Audio::Mix(float* data, uint32& numSamples)
...
-					assert(guardBuffer[i] == 0);
-				}
+					// Clearn per-channel data (and guard buffer in debug mode)
+					memset(tempData, 0, sizeof(float) * (2 * m_sampleBufferLength + guardBand));
+					item->Process(tempData, m_sampleBufferLength);
+#if _DEBUG
+					// Check for memory corruption
+					for (uint32 i = 0; i < guardBand; i++)
+					{
+						assert(guardBuffer[i] == 0);
```
Comment (BonusPlay) [inline Audio/src/Audio.cpp]: `guardBuffer` is not defined. Addig back L#19 fixes the problem.

File: Audio/src/Audio.cpp [hunk 5]
```diff
@@ -32,42 +30,43 @@ void Audio::Mix(float* data, uint32& numSamples)
...
 #if _DEBUG
-				// Check for memory corruption
-				for (uint32 i = 0; i < guardBand; i++)
-				{
-					assert(guardBuffer[i] == 0);
-				}
+					// Check for memory corruption
+					for (uint32 i = 0; i < guardBand; i++)
+					{
+						assert(guardBuffer[i] == 0);
```
Comment (BonusPlay) [inline Audio/src/Audio.cpp]: `guardBuffer` is not defined. Addig back L#19 fixes the problem.

File: Main/src/Camera.cpp (hunk 3 above)
Comment (T045T) [reply Main/src/Camera.cpp]: figured it out: This is the calculation for the camera near clipping plane, and it's wrong (the third element of that `Vector3` should be `1.0f`).

If that value was used in the `CreatePerspective` call below, the game crashes, so instead there was a constant of `0.1f`

**Comment by T045T** [comment]
Thanks for the in-depth review!

I think I addressed all of your requests :)
Btw, on macOS at least, the number of warnings was in the single digits... were the 21k warnings from the Windows build?

**Pull request merged**